crypto/threads_pthread.c

```diff
@@ -408,6 +408,13 @@ static struct rcu_qp *update_qp(CRYPTO_RCU_LOCK *lock, uint32_t *curr_id)
     ATOMIC_STORE_N(uint32_t, &lock->reader_idx, lock->current_alloc_idx,
                    __ATOMIC_RELAXED);
 
+    /*
+     * this should make sure that the new value of reader_idx is visible in
+     * get_hold_current_qp, directly after incrementing the users count
+     */
+    ATOMIC_ADD_FETCH(&lock->qp_group[current_idx].users, (uint64_t)0,
```

Comment:
I think current_idx is not in sync with `lock->reader_idx` is that intentional? I would expect to ue `lock->reader_idx` is here so we will be operating on the same address as ATOMIC_ADD_FETCH() in `get_hold_current_qp()` is referring to. Though I'm not sure it matters.

Reply:
yes that is the intention `get_hold_current_qp` shall notice that it's `qp_idx` is the wrong one and try the next one.

Reply:
and yes it matters.  In the case where the race condition can happen this will be the same atomic.